### PR TITLE
[sdformat10] no absolute paths

### PR DESCRIPTION
**Pull request by autoantwort**
For #20469

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for the PR!

File: ports/sdformat10/vcpkg.json [hunk 1]
```diff
@@ -1,13 +1,21 @@
 {
   "name": "sdformat10",
   "version-string": "10.0.0",
```
Comment (PhoebeHui) [inline ports/sdformat10/vcpkg.json:3]:
```suggestion
  "version": "10.0.0",
```

**Review by PhoebeHui — APPROVED** [review]
Thanks for your updates!

File: ports/sdformat10/portfile.cmake [hunk 2]
```diff
@@ -14,28 +13,32 @@ get_filename_component(RUBY_PATH ${RUBY} DIRECTORY)
...
 # Remove debug files
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include
-                    ${CURRENT_PACKAGES_DIR}/debug/lib/cmake
-                    ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include"
+                    "${CURRENT_PACKAGES_DIR}/debug/lib/cmake"
+                    "${CURRENT_PACKAGES_DIR}/debug/share")
+
+# only used internally in sdf::findFile
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/sdformat-10.0/sdf/sdf_config.h" "#define SDF_SHARE_PATH \"${CURRENT_PACKAGES_DIR}/share/\"" "")
```
Comment (strega-nil-ms) [inline ports/sdformat10/portfile.cmake:39]: Same thing as in sdformat9; can you look into removing the dependency in findFile on SDF_SHARE_PATH?

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]

**Review by PhoebeHui — APPROVED** [review]
Thanks for your updates!

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**